python/GafferSceneUI/CustomAttributesUI.py

```diff
@@ -156,3 +159,130 @@ def __attributePopupMenu( menuDefinition, plugValueWidget ) :
+
+##########################################################################
+# PlugCreationWidget menu extensions
+##########################################################################
+
+def __addFromAffectedMenuDefinition( menu ) :
+
+	plugCreationWidget = menu.ancestor( GafferUI.PlugCreationWidget )
+	node = plugCreationWidget.plugParent().node()
+	assert( isinstance( node, GafferScene.FilteredSceneProcessor ) )
+
+	pathMatcher = IECore.PathMatcher()
+	GafferScene.SceneAlgo.matchingPaths( node["filter"], node["in"], pathMatcher )
+
+	return __addFromPathsMenuDefinition( menu, pathMatcher.paths() )
+
+def __addFromSelectedMenuDefinition( menu ) :
+
+	plugCreationWidget = menu.ancestor( GafferUI.PlugCreationWidget )
+
+	return __addFromPathsMenuDefinition(
+		menu,
+		GafferSceneUI.ScriptNodeAlgo.getSelectedPaths(
+			plugCreationWidget.plugParent().ancestor( Gaffer.ScriptNode )
+		).paths()
+	)
+
+def __addFromPathsMenuDefinition( menu, paths ) :
+
+	plugCreationWidget = menu.ancestor( GafferUI.PlugCreationWidget )
+	node = plugCreationWidget.plugParent().node()
+
+	attributes = {}
+	for path in paths :
+		attr = node["in"].fullAttributes( path, withGlobalAttributes = True )
+		attributes.update( attr )
+	existingNames = { plug["name"].getValue() for plug in plugCreationWidget.plugParent() }
+
+	attributes = dict( sorted( attributes.items() ) )
+
+	result = IECore.MenuDefinition()
+	for key, value in attributes.items() :
+		result.append(
+			"/" + key,
+			{
+				"command" : functools.partial( __createPlug, name = key, value = value ),
+				"active" : key not in existingNames
+			}
+		)
+
+	if not len( result.items() ) :
+		result.append(
+			"/No Attributes Found", { "active" : False }
+		)
+		return result
+
+	return result
+
+def __createPlug( menu, name, value ) :
+
+	plugCreationWidget = menu.ancestor( GafferUI.PlugCreationWidget )
+	plugCreationWidget.createPlug(
+		Gaffer.PlugAlgo.createPlugFromData( "plug0", Gaffer.Plug.Direction.In, Gaffer.Plug.Flags.Default, value ),
+		name = name
+	)
+
+def __plugCreationMenu( menuDefinition, widget ) :
+
+	if not Gaffer.Metadata.value( widget.plugParent(), "ui:scene:acceptsAttributes" ) :
+		return
+
+	menuDefinition.prepend( "/FromPathsDivider", { "divider" : True } )
+
+	menuDefinition.prepend(
+		"/From Selection",
+		{
+			"subMenu" : __addFromSelectedMenuDefinition
+		}
+	)
+
+	menuDefinition.prepend(
+		"/From Affected",
+		{
+			"subMenu" : __addFromAffectedMenuDefinition
+		}
+	)
+
+GafferUI.PlugCreationWidget.plugCreationMenuSignal().connect( __plugCreationMenu )
+
+##########################################################################
+# PlugCreationWidget drag & drop extension
+##########################################################################
+
+def __filteredAttributes( widget, dragDropEvent ) :
+
+	attributes = GafferSceneUI.SceneInspector.draggedAttributes( dragDropEvent )
+	if attributes is None :
+		return None
+
```

Comment:
When dragging a selection from the Scene Inspector that doesn't include attributes, we end up with the "Attributes added already" popup warning from `__attributesDropHandler()`. `SceneInspector.draggedAttributes()` is going to return `{}` in that case, rather than `None`.

We run into the same thing with dropping a selection that doesn't include options in `CustomOptionsUI.py`.

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/6726/commits/0b0ed956dbca7fea4ff58ae49d745a867c3806fe. I was in two minds about whether or not to change the handler to say "not an attribute" or let it fall through to the generic handler. I went with the latter, partly because the SceneInspector doesn't recognise global attributes yet, so the generic handler is the only way of accessing them.

> We run into the same thing with dropping a selection that doesn't include options in CustomOptionsUI.py.

Applied the same fix in https://github.com/GafferHQ/gaffer/pull/6726/commits/6fdf96c964195282d40be4f18abe4c7e9ca857ec, although there might be more justification for taking the alternative route with this one.